### ISSUE TITLE
Handset speaker not working on CAT S22

### ISSUE THREAD

**Issue by dle-baresip** [juha-h/baresip-studio#459]
I'm testing baresip on this new-to-me phone... Setup and networking work fine.  However, I get no audio through the default handset speaker.  It doesn't matter whether I am making or receiving a call: the phone is just silent.  If I switch to speaker phone mode, audio comes through.  When I switch back, again nothing.  

It's not a hardware fault: the handset speaker works when I use Linphone as well as the default Android dialer.  The mic seems to work in all cases (although there's quite a bit of feedback in speaker phone mode).  

This phone runs Andoid Go 11, and I do see audio-related errors in the [ADB log](https://github.com/juha-h/baresip-studio/files/14781359/baresip.log.sanitized.txt): 

`android.hardware.audio.service: Failed to fetch the lookup information of the device 0000000E
`

**Comment by juha-h** [comment]
Yes, there is many audio related error messages in the logcat.  There is difference on how audio focus is requested in baresip and Linphone.  Linphone has:
```
 .setUsage(AudioAttributesCompat.USAGE_MEDIA)
```
whereas baresip has
```
.setUsage(AudioAttributesCompat.USAGE_VOICE_COMMUNICATION)
```
That could explain the issue you experienced.  However,  I don't want to change it.  There was some good reason why I'm using USAGE_VOICE_COMMUNICATION.

Also, I just tested with Pixel 7 Android emulator and audio worked fine. So I'm afraid that you are out of luck with baresip on your device.

**Comment by dle-baresip** [comment]
I'm not sure that audio focus is the issue.  The logs indicate that focus is only requested once at the beginning of the call and abandoned once at the end of the call.  Baresip reports the result of the focus request as "granted", which I assume means success.  That same audio focus would apply in both the handset and speaker phone modes.  So I believe the focus logic is fine. 

Rather, I suspect this error stems from an audio device routing/ configuration issue.  I don't think baresip calls `set_voice_volume()' directly, but note the error on line 460: 

`DeviceHAL: Error from HAL Device in function set_voice_volume: Operation not permitted`

Also, I see that `setSpeakerPhone()` has different logic for `VERSION.SDK_INT >= 31`.  My phone running Android 11 may be following a different code path than your emulator.  However, that's not necessarily the whole issue either because I had an old phone running Andoid 6 where both audio paths worked fine.

**Comment by juha-h** [comment]
Yes, looks like focus request is not causing the issue.  baresip sets call volume only if Default Call Volume is set in Audio Settings and it is set the same way no matter what API level the device has.  Regarding setSpeakerPhone, if API level is below 31, setting is very simple: `am.isSpeakerphoneOn = enable`, which should be OK.

I have Nokia 3.1 device running pure Android 10.  Just tested and there was no problems with audio either with or w/o Default Call Vol set.

**Comment by dle-baresip** [comment]
Changing the default volume setting does not seem to have any effect for me.  

In my cursory review, it looks like Linphone is using the `setAudioRoute` API and the associated constants (such as `ROUTE_EARPIECE`).  I'm not an Android expert, but this appears to be an even older API that pre-dates `isSpeakerphoneOn`.  

Why would baresip use `am.isSpeakerphoneOn = enable` instead of `am.setSpeakerphoneOn(enable)`?  Is there some magical property translation that turns a getter into a setter?

**Comment by juha-h** [comment]
Yes, there is.  If I write in Android Studio
```
am.setSpeakerphoneOn(enable)
```
it tells:
```
Use of setter method instead of property access syntax
```
and rewrites it to
```
am.isSpeakerphoneOn = enable
```

**Comment by juha-h** [comment]
I made one more test using my Nokia 3.1 Android 10 device and turning speaker phone on/off during call worked fine.

**Comment by juha-h** [comment]
Cannot reproduce the issue on my devices or emulators.
